extensions/common/gcp/gcp-core/src/test/java/org/eclipse/edc/gcp/bigquery/BigQueryServiceImplTest.java

```diff
@@ -0,0 +1,364 @@
+    }
+
+    private void testRunSinkQuery(boolean succeeds) throws DescriptorValidationException, IOException, InterruptedException, ExecutionException, TimeoutException {
+        var parts = new ArrayList<Part>();
+        parts.add(getPart(TEST_FIELDS, Arrays.asList(TEST_ROW_1, TEST_ROW_2)));
+        var partsCount = parts.stream().mapToLong(o -> o.size()).sum();
+
+        if (succeeds) {
+            ApiFuture<AppendRowsResponse> future = mock();
+            AppendRowsResponse response = mock();
+            when(future.isDone()).thenReturn(true);
+            when(future.cancel(anyBoolean())).thenReturn(false);
+            when(future.isCancelled()).thenReturn(false);
+            doNothing().when(future).addListener(any(Runnable.class), any(Executor.class));
+            when(future.get()).thenReturn(response);
+            when(future.get(anyLong(), any())).thenReturn(response);
+            when(streamWriter.append(any())).thenReturn(future);
+        } else {
+            when(streamWriter.append(any())).thenThrow(new IOException());
+        }
+        when(streamWriter.isClosed()).thenReturn(false);
+        when(streamWriter.getStreamName()).thenReturn("name");
+        when(writeClient.finalizeWriteStream(streamWriter.getStreamName())).thenReturn(
+                FinalizeWriteStreamResponse.newBuilder().setRowCount(partsCount).build());
+
+        var commitResponse = mock(BatchCommitWriteStreamsResponse.class);
+        when(commitResponse.hasCommitTime()).thenReturn(true);
+        when(commitResponse.getStreamErrorsList()).thenReturn(Arrays.asList());
+        when(writeClient.batchCommitWriteStreams(any(BatchCommitWriteStreamsRequest.class))).thenReturn(commitResponse);
+
+        var executorService = Executors.newSingleThreadScheduledExecutor();
+        executorService.schedule(() -> bigQueryService.testAppendSignal(), 2, TimeUnit.SECONDS);
+
+        bigQueryService.runSinkQuery(parts);
+
+        verify(streamWriter, times(parts.size())).append(any());
+        verify(streamWriter).close();
+        verify(writeClient).finalizeWriteStream("name");
+        if (succeeds) {
+            verify(writeClient).batchCommitWriteStreams(any(BatchCommitWriteStreamsRequest.class));
+        } else {
+            // If an error occurred while appending rows, then data should not be committed.
+            verify(writeClient, never()).batchCommitWriteStreams(any(BatchCommitWriteStreamsRequest.class));
+        }
+    }
+
+    @Test
+    void testRunSinkQuerySucceeds() throws DescriptorValidationException, IOException, InterruptedException, ExecutionException, TimeoutException {
+        testRunSinkQuery(true);
+    }
+
+    @Test
+    void testRunSinkQueryFailsNoCommit() throws DescriptorValidationException, IOException, InterruptedException, ExecutionException, TimeoutException {
+        testRunSinkQuery(false);
+    }
+
+    private void testRunSourceQuery(String sinkType) throws InterruptedException, IOException {
```

Comment:
nit: method ordering should be such, that test methods go first, then protected methods then private methods.

Reply:
Updated ordering